### ISSUE TITLE
Fast commands with timeout always hit timeout

### ISSUE THREAD

**Issue by nicholascapo** [amoffat/sh#344]
Fast commands with timeout always hit timeout:

```python
import sh
sh.bash('-c', 'exit 0', _timeout=5)
```

```bash
Exception in thread background thread for pid 27690:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "sh/sh.py", line 2159, in background_thread
    timeout_fn()
  File "sh/sh.py", line 1913, in timeout_fn
    self.signal(ca["timeout_signal"])
  File "sh/sh.py", line 2004, in signal
    os.kill(self.pid, sig)
OSError: [Errno 3] No such process

Traceback (most recent call last):
  File "./repro.py", line 4, in <module>
    sh.bash('-c', 'exit 0', _timeout=5)
  File "sh/sh.py", line 1245, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "sh/sh.py", line 633, in __init__
    self.wait()
  File "sh/sh.py", line 648, in wait
    raise TimeoutException(-exit_code)
sh.TimeoutException
```

`git bisect start 1.12.5 1.11` with the above repro script:
4d6594b65891e95131f5d36593b9f423f5d80535 is the first bad commit
commit 4d6594b65891e95131f5d36593b9f423f5d80535
Author: Andrew Moffat <andrew.robert.moffat@gmail.com>
Date:   Thu Dec 1 00:14:47 2016 -0600

    wait on the bg thread, like we do the other threads

**Comment by amoffat** [comment]
it's the same underlying issue as reported here https://github.com/amoffat/sh/issues/340.  the temporary workaround is to remove the timeout.  i should have a fix out later today

**Comment by amoffat** [comment]
the issue is fixed on the latest release, 1.12.6
